### PR TITLE
PROPOSAL: Make other file types to also expose their text source.

### PR DESCRIPTION
**Pull request by renoirb**
**Idea** Make other file types to also expose their text source.

## Types of changes

- New feature (a non-breaking change which adds functionality)

## Description

In some situations, we might want to have the raw source to use with own syntax highlighter, say for a static site.

The `$content([...], { text: true })` only works for Markdown. This should allow for other types too.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes (if not applicable, please state why)

### CODE REVIEW

File: packages/content/lib/database.js [hunk 1]
```diff
@@ -200,7 +200,12 @@ class Database extends Hookable {
     // Collect data from file
     let data = []
     try {
-      data = await parser(file.data)
+      const text = await file.data
```
Comment (renoirb) [inline packages/content/lib/database.js:203]: That's assuming data is an `async` getter and not the parser that's the asynchronous.

**Comment by benjamincanac** [comment]
@renoirb What is the use-case here? If you want to do this, it would be best to return the text content from the parsers, see how it's done for markdown: https://github.com/nuxt/content/blob/dev/packages/content/parsers/markdown/index.js#L107.

**Comment by renoirb** [comment]
> @renoirb What is the use-case here? 

The use-case I had is when I want to display the raw source. The markdown source is available in raw probably for the editor. In my case, I wanted the raw YAML source so that I could show it in a colorized code block.

**Comment by renoirb** [comment]
Hey @benjamincanac do you want me to clarify the use-case?

**Comment by renoirb** [comment]
FC816AE4-D442-4985-90CE-DA9BA4675A6B

**Comment by Tahul** [comment]
Hey @renoirb 😄

I think you could achieve the exact result from your screenshot by using code block in @nuxt/content v2.

YML Syntax highlighting will be provided by [Shiki](https://github.com/shikijs/shiki)!

I am closing this in favor of v2 but feel free to re-open if you want to continue the work on this PR.

**Comment by renoirb** [comment]
I shall have a look. I can't wait to see how nuxt content changed.

**Pull request closed without merging**